### ISSUE TITLE
Consider to add `class` as an additional configuration in `space-before-blocks` rule

### ISSUE THREAD

**Issue by almirfilho** [eslint/eslint#4089]
What do you guys think about adding an additional config in `space-before-blocks` rule for ES6 classes?
Actually there are only two possible options: `functions` and `keywords`. However, it might be desirable to consider classes as a separated config as well. For instance, I might want my code to follow the rule as `{"keywords": false, "functions": false, "classes": true}`:

```javascript
class Foo extends Bar {
  someMethod(){
    if(something){
      // do something
    }
  }
}
```

Thank you!

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
Which version of eslint are you using?
This has already been added: https://github.com/eslint/eslint/pull/3066

**Comment by almirfilho** [comment]
Hi @gyandeeps. I think #3066 just considers class statements to be linted with the rule `space-before-blocks`, but there's no way to configure spaces before classes separately. As we can see in [the docs](http://eslint.org/docs/rules/space-before-blocks), there're only options for `functions` and `keywords`, but not for classes.

**Comment by gyandeeps** [comment]
Ah, I see. I think I forgot to add that during that enhancement.
@nzakas @ilyavolodin Do we flag this as a bug or enchancement?

**Comment by gyandeeps** [comment]
@almirfilho Would be willing to work on this fix?

**Comment by ilyavolodin** [comment]
This is an enhancement as #3811 was fixing a bug where classes were not linted at all using this rule. This request is to add new configuration option.
I'm not using es6, so my opinion is not really matter that much, but I would consider classes to be falling under `function` flag. Not sure I see a huge need to add another option specifically for them.

**Comment by gyandeeps** [comment]
@nzakas @ilyavolodin What should be done here?

**Comment by nzakas** [comment]
Let's add the option. It's nice and explicit.
